quality_check/10-test_compendium_PLIER.Rmd

```diff
@@ -0,0 +1,308 @@
+---
+title: "Run PLIER on test compendium"
+output: 
+  html_notebook:
+    toc: TRUE
+    toc_float: TRUE
+author: J. Taroni for CCDL
+date: 2019
+---
+
+Do we "get more biology" out of the combined technology test compendium than we
+get out of each of the technologies individually?
+
+We'll use the metrics we developed as part of the MultiPLIER project here.
+Specifically, do we get higher pathway coverage for a model trained on
+the dual-technology test compendium vs. those trained on the individual
+technology compendia?
+
+## Setup
+
+#### Functions
+
+Source the most recent version of the MultiPLIER utils from Github
+
+```{r}
+source("https://github.com/greenelab/multi-plier/raw/7c56a2867f9f08f7bcd3617875c7c2dbe886cbeb/util/plier_util.R")
+# we only want GetPathwayCoverage
+rm(list = setdiff(ls(), "GetPathwayCoverage"))
+```
+
+```{r}
+`%>%` <- dplyr::`%>%`
+```
+
+```{r}
+# we'll train 3 PLIER models: matrix with both technologies, microarray samples
+# only, and RNA-seq samples only
+PLIERWrapper <- function(exprs_mat, common_genes, pathway_mat, seed = 123) {
+  
+  # set seed for reproducibility
+  set.seed(seed)
+  
+  # z-score the expression data
+  row_norm_exprs <- PLIER::rowNorm(exprs_mat[common_genes, ])
+  
+  # how many "significant PCs" via the elbow method -- we use this to set k when
+  # training a PLIER model
+  num_pcs <- PLIER::num.pc(row_norm_exprs)
+  
+  return(PLIER::PLIER(data = row_norm_exprs[common_genes, ], 
+                      priorMat = pathway_mat[common_genes, ],
+                      k = round(num_pcs * 1.3)))
+}
+```
+
+#### Libraries
+
+```{r}
+# PLIER installation
+# BiocManager::install("qvalue", update = FALSE)
+# devtools::install_github('wgmao/PLIER',
+#                          ref = 'a2d4a2aa343f9ed4b9b945c04326bebd31533d4d',
+#                          dependencies = TRUE)
+# install.packages("cowplot")
+```
+
```

Comment:
Do you usually combine these into one chunk? Now I don't remember. Suppose it doesn't matter.